src/docs/asciidoc/reference/architecture/code-structure.adoc

```diff
@@ -1,2 +1,150 @@
+You can define a lot of configurations according to the link:{typescript-conf-url}[TypeScript documentation].
+
+For the proper functioning of the *ZetaPush application*, the `tsconfig.json` file needs to have some minimal configuration like explained below :
+
+include::{docdir}/common/code/minimal-tsconfig-needed.adoc[]
+
+.Customize my tsconfig.json
+
+In this case of you need to customize your `tsconfig.json` file, for example to use classes only declared in ES6 (es2015), you can add this in the `compilerOptions`. Of course, you can add any properties provided by link:{typescript-conf-url}[TypeScript].
+
+.Customized tsconfig.json
+[source,json,linenums]
+----
+{
+  "compilerOptions": {
+    "lib": [
+      "es2015"
+    ],
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true
+  }
+}
+----
+
+=== worker folder
+
+As seen above, the `worker` folder contains the business logic of your application. It can also called your "back code".
+This is in this part that you will create your business logic and interact with the _built-in cloud services_ provided by ZetaPush (User management, file system, database, notifications, etc...).
+
+By default (When the application is generated by the _CLI_), the `worker` folder contains only one file, called `index.ts`.
+
+include::{docdir}/common/code/generated-worker-code.adoc[]
+
+.Develop your worker in several ways
+[NOTE]
+====
+In this example we see a worker that is created with only one class. You will see in the <<reference.adoc#develop_custom_cloud_services, cloud service development section>> how to create a worker in several ways.
+====
+
+.Don't need to transpile TypeScript
+[NOTE]
+====
+Even if your worker code is in `TypeScript` you don't need to transpile it to `JavaScript`. Indeed, when you run or push your code, it will be launched with link:{ts-node-url}[`ts-node`]. +
+`ts-node` take care of the transpilation, you can also configure this via the `tsconfig.json` file.
+====
+
+=== front folder
+
+The `front` folder contains your front code.
+So you need to have your `index.html` (or other html file if you want) at the root of this folder.
+
+.Use frameworks
+[NOTE]
+====
+You can integrate your favorite framework with ZetaPush.
+For now, we give you documentation for integration with the most famous frameworks : <<reference#develop_with_angular, Angular>>,
+<<reference#develop_with_react, React>> and <<reference#develop_with_vue, Vue.js>>.
+====
+
+=== application.json
+
+This file is *optional* in a *ZetaPush application*. For now you just need to use it when you use the <<developer-manual.adoc#user-management-in-application, StandardUserWorkflow>>.
+Of course, you can also use it without using the <<developer-manual.adoc#user-management-in-application, StandardUserWorkflow>>.
+
+include::{docdir}/common/code/application-json-standard-user-workflow.adoc[]
```

Comment:
I don't think we should place example with all properties of StandardUserWorkflow here. Moreover most of them are optional now. I think we should indicate that some builtin cloud services are using it for configuration and that the developer can also use it (with a link to show how to define custom properties and to use it in code)